src/array.cr

```diff
@@ -852,6 +827,10 @@ class Array(T)
     @size = size.to_i
   end
 
+  def bytesize
+    sizeof(T) * @size
+  end
```

Comment:
What are the usecases for this method on `Array`? Shouldn't we encourage usage of Slice etc for these?